packages/components/pie-input/test/component/pie-input.spec.ts

```diff
@@ -490,6 +490,59 @@ test.describe('PieInput - Component tests', () => {
                 expect(await page.evaluate(() => document.querySelector('pie-input')?.value)).toBe('foo');
             });
         });
+
+        test.describe('disabled', () => {
+            test('should not disable the component if the prop is not provided', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieInput, {});
+
+                // Act
+                const input = await component.locator('input');
+
+                // Assert
+                expect(input).not.toBeDisabled();
+            });
+
+            test('should disable the component if the prop is provided', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieInput, {
+                    props: {
+                        disabled: true,
+                        value: 'test',
+                    } as InputProps,
+                });
+
+                // Act
+                const input = await component.locator('input');
+
+                // Assert
+                expect(input).toBeDisabled();
+            });
+
+            test('should not be able to focus the component when disabled', async ({ page }) => {
+                // Arrange
+                await page.setContent('<pie-input type="text" disabled></pie-input>');
+
+                // Act
+                const input = await page.locator('pie-input');
+                await input.focus();
+
+                // Assert
+                expect(input).not.toBeFocused();
+            });
+
+            test('should still be able to focus the component when not disabled', async ({ page }) => {
```

Comment:
Should this test be here because it falls under the describe of `disabled`?

Reply:
I see it as a variation of the value for the `disabled` prop, so I think it does belong here 👍🏻

Reply:
OK. How about the below (appreciate it might be a nit picky so can keep as is, i just think it would read better/clearer imo with the below).

```javascript
describe('`disabled`')
  // other tests
   describe('when not provided')
      test('should still be able to focus the component')
```